FileTypeIcon enhancements (#1789)
pnp/sp-dev-fx-controls-react: docs/documentation/docs/controls/FileTypeIcon.md

| path | string | no | Path to the document. If this is provided, the control will use the file extension to display the corresponding icon. |
| size | ImageSize | no | This is a property that only needs to be used when the type is set to image. It allows you to specify the image size. small (16px), normal (20px), medium (48px) and large (96px) are possible. Use the **ImageSize** enum to get the list of available images sizes. |
| type | IconType | yes | This property specifies is you want to use the icon font or image. Use the **IconType** enum to get the list of available icon types. |

> | onClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is clicked. |

Reviewer: ```suggestion
| onClick | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the icon is clicked. |
```

> | onDoubleClick | React.MouseEvent\<HTMLElement> | no | Event triggered when the icon is double clicked. |

Reviewer: ```suggestion
| onDoubleClick | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the icon is double clicked. |
```

> | onMouseEnter | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor enter the icon. |

Reviewer: ```suggestion
| onMouseEnter | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor enters the icon (without event bubbling). |
```

> | onMouseLeave | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor leave the icon. |

Reviewer: ```suggestion
| onMouseLeave | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor leaves the icon. |
```

> | onMouseOver | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse cursor is over the icon. |

Reviewer: ```suggestion
| onMouseOver | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse cursor enters the icon (with event bubbling). |
```

> | onMouseUp | React.MouseEvent\<HTMLElement> | no | Event triggered when the mouse button is released after clicked on the icon. |

Reviewer: ```suggestion
| onMouseUp | React.MouseEvent&lt;HTMLElement&gt; | no | Event triggered when the mouse button is released after clicked on the icon. |
```
